src/Interpreters/ProcessList.h

```diff
@@ -329,6 +331,9 @@ class ProcessList : public ProcessListBase
 
     mutable std::condition_variable have_space;        /// Number of currently running queries has become less than maximum.
 
+    /// Do not allow to modify processes list during killAllQueries operations
```

Comment:
It does not allow modifications in `~ProcessListEntry`, consider throwing `LOGICAL_ERROR` or adding `chassert(!block_processes)` in other places where the list is modified